samples/reflection/CMakeLists.txt

```diff
@@ -9,23 +7,19 @@ include(CTest)
 # sample, and the actual C source of this project.
 #
 
-set(INC_DIR "${PROJECT_SOURCE_DIR}/include")
 set(GEN_DIR "${CMAKE_CURRENT_BINARY_DIR}/generated")
 set(FBS_DIR "${PROJECT_SOURCE_DIR}/samples/monster")
 
-include_directories("${GEN_DIR}" "${INC_DIR}")
-
-add_custom_target(gen_monster_bfbs ALL)
-add_custom_command (
-    TARGET gen_monster_bfbs
-    COMMAND cmake -E make_directory "${GEN_DIR}"
-    COMMAND flatcc_cli --schema -o "${GEN_DIR}" "${FBS_DIR}/monster.fbs"
-    DEPENDS flatcc_cli "${FBS_DIR}/monster.fbs"
+flatcc_generate_sources(
+    NAME reflection_sample
+    SCHEMA_FILES "${FBS_DIR}/monster.fbs"
+    BINARY_SCHEMA
+    OUTPUT_DIR "${GEN_DIR}"
 )
+
 add_executable(bfbs2json bfbs2json.c)
-add_dependencies(bfbs2json gen_monster_bfbs)
-target_link_libraries(bfbs2json flatccrt)
+target_link_libraries(bfbs2json PRIVATE flatcc::runtime flatcc_generated::reflection_sample)
 
 if (FLATCC_TEST)
-    add_test(bfbs2json bfbs2json${CMAKE_EXECUTABLE_SUFFIX} ${GEN_DIR}/monster.bfbs)
+    add_test(NAME bfbs2json COMMAND bfbs2json "${GEN_DIR}/monster.bfbs")
 endif()
```

Comment:
The flatcc samples and tests are only built when `FLATCC_TEST` is true.
So I think you can do `add_test` unconditionally.

Reply:
Yes a noticed that as well, in various places all around.